src/mscorlib/src/System/MarshalByRefObject.cs

```diff
@@ -35,10 +35,10 @@ namespace System {
 [System.Runtime.InteropServices.ComVisible(true)]
     public abstract class MarshalByRefObject 
     {
-#if FEATURE_REMOTING    
         private Object __identity;        
```

Comment:
You either need to keep this under `FEATURE_REMOTING` ifdef; or remove the matching ifdefs in VM. I think that keeping this under `FEATURE_REMOTING` ifdef is better.

The test failures are caused by this problem.